### DISCUSSION TITLE
glTF loads in model-viewer, crashes in scene-viewer

### DISCUSSION THREAD

**Discussion by MajorMeerkatThe3rd** [google/model-viewer#3023] [Q&A] [answered]
Hello,

great work that you are doing here. I have a glTF that loads fine on https://modelviewer.dev/editor/ and can be displayed via the QR Code that is created there.

When I try the same glTF with the scene-viewer, the AR App completely breaks, like badly. What could be the cause of that?

My code to reproduce:
```
const a = document.createElement('a');
a.href = 'intent://arvr.google.com/scene-viewer/1.0?file=https://shapediverviewer.s3.amazonaws.com/v3/examples/gltf/2.0/d863cca30573359a2058d61eb4a8e6085874600ce6da02ca207988421554c09d3406a973dae2290906e6d3540fcbf8f85ed4943230e22881fdabfd3764acb72f66b685bb6c109b06ffe341ad5f59762ae58bd5fd472bca3e3015e9854a251454c2501730ed34cc9361f55ce.glb&mode=3d_preferred#Intent;scheme=https;package=com.google.ar.core;action=android.intent.action.VIEW;end;'
document.body.appendChild(a);
a.click();
```

This glTF is created by our internal glTF-converter. This is the first time we encountered this issue. Help would be very much appreciated.

Cheers

**Comment by elalish** [comment]
@AdrianAtGoogle Another problem GLB for scene viewer; any thoughts?  You can download it from his file URL above.

**Comment by AdrianAtGoogle** [comment, marked as answer]
Neat - thanks for reporting the issue.

SceneViewer isn't handling the case correctly where a mesh elides specifying indices.  

I have a fix for it submitting to head shortly, it'll be a couple of weeks before this shakes all the way out to the mobile app.

**Comment by MajorMeerkatThe3rd** [comment]
Thank you for the fast fix! Awesome support.

Completely understandable that this takes some time to get to the mobile app.
